hdfs/client.py

```diff
@@ -430,8 +479,8 @@ def write(self, hdfs_path, data=None, overwrite=False, permission=None,
     def consumer(_data):
       """Thread target."""
       self._request(
-        method='POST' if append else 'PUT',
```

Comment:
These changes shouldn't be needed once the above logic change has been moved out of the `_request` method.